### DISCUSSION TITLE
可以在HY2增加指定特定域名使用特定的 DNS 服务器这个功能吗？

### DISCUSSION THREAD

**Discussion by ghost** [apernet/hysteria#855] [Ideas]
根据 Hysteria 2 文档的内容，要设置特定网站（如 openai.com）的流量通过指定的 DNS 服务器（8.8.8.8），你需要在 HY2 的 server.yaml 文件中配置相应的 ACL 规则。然而，重要的是要注意，根据 Hysteria 2 的文档，它似乎不支持直接通过 ACL 配置特定的 DNS 服务器来处理特定域名的 DNS 请求。这是GPT的回答。想请问作者大大什么时候可以添加这个功能。
注：resolver 部分指定用于解析客户端请求中的域名的 DNS 服务器。，并不能实现dns分流功能

**Comment by yleaway** [comment]
作者文档里有的呀
在config.yaml最后添加
acl:
 inline: 
   - default(geosite:openai, *, 8.8.8.8)

我试了下劫持奈飞到解锁DNS，貌似没啥问题
btw:作者的说明文档写的真的好

**Reply by haruue** [reply]
```yaml
acl:
 inline: 
   - default(geosite:openai, *, 8.8.8.8)
```

这个操作实际上是把所有本该发给 openai 域名的流量劫持到 `8.8.8.8` ， 不是用 `8.8.8.8` 解析 openai 地址。

此外， 大部分 netflix 解锁 DNS 实现（包括 apernet 以前那个）， 是用同一个地址作为 DNS 服务器和 sniproxy， 这种情况下用这个 DNS 解析 netflix 的域名， 获得的地址也会是 DNS 服务器自身， 最终的目标也是劫持 netflix 流量到 DNS 服务器上开着的 sniproxy。 你这个用例可以说是歪打正着了， 但是不是楼主想问的。

**Reply by ghost** [reply]
确实不是我想问的，我后面发现通过dnsmasq可以实现dns分流

``` 
配置 dnsmasq
sudo nano etc/dnsmasq.conf

# 指定默认DNS服务器
server=8.8.8.8
server=8.8.4.4

# 为特定域名使用特定的DNS服务器
server=/openai.com/1.1.1.1

```

**Reply by FFFKLD** [reply]
> 确实不是我想问的，我后面发现通过dnsmasq可以实现dns分流
> 
> ```
> 配置 dnsmasq
> sudo nano etc/dnsmasq.conf
> 
> # 指定默认DNS服务器
> server=8.8.8.8
> server=8.8.4.4
> 
> # 为特定域名使用特定的DNS服务器
> server=/openai.com/1.1.1.1
> ```

可以咨询一下大佬您一些关于DNS分流方面的问题吗，这是我的QQ：807290915，可有偿，谢谢大佬！
